File: android/src/main/kotlin/tech/soit/quiet/player/PlayMode.kt
package tech.soit.quiet.player

/**
 *
 * PlayMode of MusicPlayer
 *
 * @author 杨彬
 */
sealed class PlayMode(
    val rawValue: Int
) {
    companion object {
        fun valueOf(value: Int): PlayMode = when (value) {
            0 -> Shuffle
            1 -> Single
            2 -> Sequence
            else -> Undefined(value)
        }
    }

    //随机播放
    object Shuffle : PlayMode(0)

    //单曲循环
    object Single : PlayMode(1)

    //列表循环
    object Sequence : PlayMode(2)

    class Undefined(value: Int) : PlayMode(value)

}

File: macos/MusicPlayerPlugin/PlayMode.swift
//
// Created by yangbin on 2020/2/22.
//

import Foundation

enum PlayMode: Equatable {
    case shuffle
    case single
    case sequence
    case undefined(value: Int)
}

extension PlayMode {

    static func from(rawValue: Int) -> PlayMode {
        switch (rawValue) {
        case 0:
            return PlayMode.shuffle
        case 1:
            return PlayMode.single
        case 2:
            return PlayMode.sequence
        default:
            return PlayMode.undefined(value: rawValue)
        }
    }

    var rawValue: Int {
        switch (self) {
        case .shuffle:
            return 0;
        case .single:
            return 1;
        case .sequence:
            return 2;
        case .undefined(let value):
            return value
        }
    }

}